changelog/12749.feature.rst

```diff
@@ -0,0 +1,3 @@
+Add :confval:`discover_imports`, when disabled (default) will make sure to not consider classes which are imported by a test file and starts with Test.
```

Comment:
Nice, changelog entry.

We also need to add documentation for that confval in `reference.rst`.


I'm not sure about the name too, but at the moment I don't have another suggestion (however we really should strive to find a more appropriate name).

Reply:
treat_imports_as_tests?
imports_are_tests?
crawl_imports?

Reply:
As per [my ](https://github.com/pytest-dev/pytest/pull/12810#issuecomment-2348621005), perhaps we should focus on being more "strict" about what we collect from modules.

`collect_?something? = "all"/"local-only"`

But we should wait to see what other maintainers think.

Reply:
Alright, now after a some hours (more than id like to admit) I now have something that works. Both for functions and classes. Let me know how we wanna handle the toggle of the feature. Have a good weekend!